UnchainedLauncherCore/src/API/A2S.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net;
+using System.Net.Sockets;
+using System.Buffers.Binary;
+
+namespace UnchainedLauncher.Core.API
+{
+    public record A2S_INFO (
+        byte ProtocolVersion, 
+        String Name,
+        String Map,
+        String Folder,
+        String Game, 
+        UInt16 GameID, 
+        byte Players,
+        byte MaxPlayers,
+        byte Bots, 
+        ServerType ServerType, 
+        Environment Environment, 
+        bool IsPublic, 
+        bool Vac
+        );
+
+    public enum ServerType : byte
+    {
+        DEDICATED = (byte)'D',
+        NONDEDICATED = (byte)'L',
+        PROXY = (byte)'P'
+    }
+
+    public enum Environment : byte
+    {
+        LINUX = (byte)'L',
+        WINDOWS = (byte)'W',
+        MAC = (byte)'M'
+    }
+
+    public static class A2S
+    {
+        const int TimeOutMillis = 1000;
+        public static async Task<A2S_INFO> InfoAsync(IPEndPoint ep)
+        {
+            try
+            {
+                return await InfoAsync_impl(ep);
+            }
+            catch(TaskCanceledException)
+            {
+                throw new TimeoutException("A2S connection timed out");
+            }
+        }
+        private static async Task<A2S_INFO> InfoAsync_impl(IPEndPoint ep)
```

Comment:
Any reason this is separated from the InfoASync method? Usually I see this kind of separation when there's recursion happening in the `_impl` method, to avoid repeating unnecessary work. But I don't see any recursion happening here

Reply:
I see this pattern throughout the code. Can the `*_impl` pattern go away? It seems unnecessary

Reply:
I'm using the impl pattern just so that I don't have to have that try-catch to convert from a cancellation exception to a timeout exception in the implementation itself. Looking back now, I think it's adding more clutter than it's worth. The conversion will probably go away anyways once I use the native timeouts instead of cancellation tokens anyways.